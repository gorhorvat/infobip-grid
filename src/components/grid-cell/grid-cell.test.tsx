import React from "react";
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import GridCell from "./grid-cell";
import CellModel from "../../models/cell-model";

describe('GridCell', () => {
  const cellParam: CellModel = {
    id: 0,
    value: 1,
    isClicked: false
  }

  function setup(isClicked: boolean) {
    return shallow(<GridCell cell={{ ...cellParam, isClicked: isClicked }} onCellClick={jest.fn()} />);
  }

  it('Renders without crashing', () => {
    // Arrange
    const wrapper = setup(false);

    // Assert
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('When cell is clicked should show value', () => {
    // Arrange
    const wrapper = setup(true);

    // Assert
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('When cell is clicked should not show value', () => {
    // Arrange
    const wrapper = setup(false);

    // Assert
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});