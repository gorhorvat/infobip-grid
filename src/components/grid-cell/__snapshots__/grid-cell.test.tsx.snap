// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridCell Renders without crashing 1`] = `
<td
  className="grid-cell grid-cell--inactive"
  key="0"
  onClick={[MockFunction]}
/>
`;

exports[`GridCell When cell is clicked should not show value 1`] = `
<td
  className="grid-cell grid-cell--inactive"
  key="0"
  onClick={[MockFunction]}
/>
`;

exports[`GridCell When cell is clicked should show value 1`] = `
<td
  className="grid-cell grid-cell--active"
  key="0"
  onClick={[MockFunction]}
>
  1
</td>
`;
